packages/expo-modules-core/android/src/main/java/expo/modules/kotlin/modules/Module.kt

```diff
@@ -12,11 +15,22 @@ abstract class Module {
   val appContext: AppContext
     get() = requireNotNull(_appContext) { "The module wasn't created! You can't access the app context." }
 
+  @Suppress("PropertyName")
+  @PublishedApi
+  internal lateinit var _coroutineScopeDelegate: Lazy<CoroutineScope>
+  val coroutineScope by _coroutineScopeDelegate
+
   fun sendEvent(name: String, body: Bundle?) {
     moduleEventEmitter?.emit(name, body)
   }
 
   abstract fun definition(): ModuleDefinitionData
+
+  internal fun cleanUp() {
+    if (_coroutineScopeDelegate.isInitialized()) {
+      coroutineScope.cancel(ModuleDestroyedException())
```

Comment:
Shouldn't it be
```suggestion
      try {
        coroutineScope.cancel(ModuleDestroyedException())
      } catch (e: IllegalStateException) {}
```
https://github.com/expo/expo/pull/16413#discussion_r817041960

or am I wrong?